presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -3160,6 +3229,46 @@ private static void validatePartitionColumns(ConnectorTableMetadata tableMetadat
         }
     }
 
+    private static void validateMaterializedViewPartitionColumns(Table viewTable, List<Table> baseTables, Map<String, Map<SchemaTableName, String>> viewToBaseColumnMap)
+    {
+        SchemaTableName viewName = new SchemaTableName(viewTable.getDatabaseName(), viewTable.getTableName());
+        if (viewToBaseColumnMap.isEmpty()) {
+            throw new PrestoException(
+                    NOT_SUPPORTED,
+                    "Materialized view must have at least one column directly defined by a base table column.");
+        }
+
+        List<Column> viewPartitions = viewTable.getPartitionColumns();
+        if (viewPartitions.isEmpty()) {
+            throw new PrestoException(NOT_SUPPORTED, "Unpartitioned materialized view is not supported.");
+        }
+        Map<SchemaTableName, List<Column>> baseTablePartitions = baseTables.stream()
+                .collect(toImmutableMap(
+                        table -> new SchemaTableName(table.getDatabaseName(), table.getTableName()),
+                        Table::getPartitionColumns));
+
+        Map<Column, Map<SchemaTableName, Column>> viewToBasePartitionMap = new HashMap<>();
+        for (Column viewPartition : viewPartitions) {
+            for (SchemaTableName baseTable : baseTablePartitions.keySet()) {
+                for (Column basePartition : baseTablePartitions.get(baseTable)) {
+                    if (viewToBaseColumnMap
+                            .getOrDefault(viewPartition.getName(), emptyMap())
+                            .getOrDefault(baseTable, "")
+                            .equals(basePartition.getName())) {
+                        viewToBasePartitionMap.put(viewPartition, ImmutableMap.of(baseTable, basePartition));
+                        break;
+                    }
+                }
+            }
+        }
+
+        if (viewToBasePartitionMap.isEmpty()) {
```

Comment:
This doesn't seem to be the right check. We probably want to start with restrictive 1:1 mapping. Also, add a java with an example at the top of this function to explain the retionale.

Reply:
We don't need this if.